platform/ios/PolyPodApp/Features/Feature.swift

```diff
@@ -11,35 +12,57 @@ class Feature {
     let thumbnail: URL?
     private let links: [String: String]
     
-    static func load(path: URL, languageCode: String?) -> Feature? {
-        let manifest = readManifest(path)
+    static func load(path: URL) -> Feature? {
+        guard let manifest = readManifest(path) else {
+            return nil
+        }
         return Feature(
             path: path,
-            manifest: manifest,
-            languageCode: languageCode
+            manifest: manifest
         )
     }
     
-    init(path: URL, manifest: FeatureManifest, languageCode: String?) {
+    init(
+        path: URL,
+        name: String?,
+        author: String?,
+        description: String?,
+        thumbnail: String?,
+        thumbnailColor: String?,
+        primaryColor: String?,
+        links: [String: String]?
+    ) {
         self.path = path
-        let userLanguage = languageCode ?? "en"
-        let translations = manifest.translations?[userLanguage]
         let id = path.lastPathComponent
         self.id = id
-        name = translations?.name ?? manifest.name ?? id
-        author = translations?.author ?? manifest.author
-        description = translations?.description ?? manifest.description
-        let primaryColor = parseColor(hexValue: translations?.primaryColor ?? manifest.primaryColor)
+        self.name = name ?? id
+        self.author = author
+        self.description = description
+        let primaryColor = parseColor(hexValue: primaryColor)
         self.primaryColor = primaryColor
-        thumbnailColor = parseColor(hexValue: translations?.thumbnailColor ?? manifest.thumbnailColor) ?? primaryColor
-        thumbnail = findThumbnail(
+        self.thumbnailColor = parseColor(hexValue: thumbnailColor) ?? primaryColor
+        self.thumbnail = findThumbnail(
             featurePath: path,
-            thumbnailPath: translations?.thumbnail ?? manifest.thumbnail
-        )
-        links = mergeLinks(
-            original: manifest.links,
-            translated: translations?.links
+            thumbnailPath: thumbnail
         )
+        self.links = links ?? [:]
+    }
+    
+    convenience init(path: URL, manifest: FeatureManifest) {
+        var links: [String: String] = [:]
+        for idx in 0..<manifest.linksCount {
+            if let link = manifest.links(at: idx) {
+                links[link.name] = link.url
+            }
+        }
```

Comment:
Yeah that also seems like something we should handle in _PlatformCoreSwift_ (if I remember the name right). Total boilerplate for interacting with core, isn't it?

Reply:
Yes and no. I wanted to change as little as possible in the definition of `Feature` class. Refactoring of which I would like to do in a separate PR.

We could search for the urls with `findUrl` directly in the `manifest.links`, without creating an intermediary dictionary.

Reply:
If you just pass a `FeatureManifest` object to `Feature` where you handle these kinds of shenanigans, we'd have to change **less** of `Feature`, no? Since it'd still get more or less the same interface as before, except now provided by _PlatformCoreSwift_. That's the whole idea.